qtconsole/console_widget.py

```diff
@@ -2183,9 +2183,9 @@ def _insert_plain_text(self, cursor, text, flush=False):
                     format = self._ansi_processor.get_format()
                     if not (hasattr(cursor,'_insert_mode') and cursor._insert_mode):
                         pos = cursor.position()
-                        remain=self._get_line_end_pos()-pos
+                        remain=self._get_line_end_pos()-pos # could be negative ! As not taking int account last \n ?
                         n=len(substring)
-                        swallow=min(n,remain)
+                        swallow=max(0,min(n,remain))
```

Comment:
```suggestion
                        swallow = max(0, min(n, remain))
```

Add spaces around operators and commas.